lib/block_view.go

```diff
@@ -7166,25 +7518,54 @@ func (bav *UtxoView) _connectAcceptNFTBid(
 	// Sanity check that the royalties are reasonable and won't cause underflow.
 	if txMeta.BidAmountNanos < (creatorRoyaltyNanos + creatorCoinRoyaltyNanos) {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: sum of royalties (%d, %d) is less than bid amount (%d)",
+			"_helpConnectNFTSold: sum of royalties (%d, %d) is less than bid amount (%d)",
 			creatorRoyaltyNanos, creatorCoinRoyaltyNanos, txMeta.BidAmountNanos)
 	}
 
 	bidAmountMinusRoyalties := txMeta.BidAmountNanos - creatorRoyaltyNanos - creatorCoinRoyaltyNanos
-
+	txn, txHash, verifySignatures := txMeta.Txn, txMeta.TxHash, txMeta.VerifySignatures
 	// Connect basic txn to get the total input and the total output without
 	// considering the transaction metadata.
 	totalInput, totalOutput, utxoOpsFromBasicTransfer, err := bav._connectBasicTransfer(
 		txn, txHash, blockHeight, verifySignatures)
 	if err != nil {
-		return 0, 0, nil, errors.Wrapf(err, "_connectAcceptNFTBid: ")
+		return 0, 0, nil, errors.Wrapf(err, "_helpConnectNFTSold: ")
 	}
 	// Append the basic transfer utxoOps to our list
 	utxoOpsForTxn = append(utxoOpsForTxn, utxoOpsFromBasicTransfer...)
 
 	// Force the input to be non-zero so that we can prevent replay attacks.
 	if totalInput == 0 {
-		return 0, 0, nil, RuleErrorAcceptNFTBidRequiresNonZeroInput
+		return 0, 0, nil, errors.Wrapf(RuleErrorAcceptNFTBidRequiresNonZeroInput, "_helpConnectNFTSold: ")
+	}
+
+	if txMeta.IsNFTBid {
+		bidAmountNanos := txMeta.BidAmountNanos
+		// Check that the bid amount is non-zero.
+		if bidAmountNanos == 0 {
```

Comment:
Is this check needed? Seems like `bidAmountNanos < prevNFTEntry.MinBidAmountNanos` has you covered.

Reply:
Going with a previous comment, it seems like bidding zero should be allowed if the person puts an NFT up for sale with a min bid of zero. It's kindof like a "race" for who can bid on the NFT first.

Reply:
Update: Nevermind. Clearly a bid of zero means you want to delete the bid, which doesn't make sense for an IsNFTBid I think...

Reply:
correct! yeah 0 is cancel my bid. unless we change this at a later date to be a -1 means cancel my bid. but that introduces a lot of changes. so I'm inclined to not allow 0 at this point in time.